docs: add proveFault interface (#21)
kroma-network/kroma-specs: specs/experimental/zk-fault-dipute-game/overview.md

## Proving Fault using zkVM

> Once the first disagreeing output root is specified, the challenger can prove that the claim is correct using a ZK

Reviewer: ```suggestion
Once the first disagreeing output root is specified, the challenger can prove that the disputed claim is incorrect using a ZK
```

3. Hash of the block C
```

#### Proving Fault

The ZK proof generated by a prover can be verified on chain.

```solidity
    /**
     * @notice Proves that a specific output is invalid using ZK proof.
     *
     * @param _pos          Position of the last valid segment.
     * @param _targetOutput The output root of target block.
     * @param _zkProof      SP1 proofs composed of points and scalars.
     */
    function proveFault(
        uint256 _pos,
        bytes32 _targetOutput,
        uint256[] calldata _zkProof
    ) external;
```

> The verify function, which will be used in `proveFault` for verification of provided ZK proof, will be implemented

Reviewer: ```suggestion
The verify function of `ZKVerifier` contract, which will be used in `proveFault` for verification of provided ZK proof, will be implemented
```

following the below interface.

```solidity
interface ZKVerifier {
    /** @notice The entrypoint for verifying the proof of ZK fault proof.
     *
     *  @param _publicValues The encoded public values.
     *  @param _proofBytes The encoded proof.
     *

>      *  @return If verification of ZK proof is passed or not.

Reviewer: ```suggestion
    /** 
     * @notice The entrypoint for verifying the ZK fault proof.
     *
     * @param _publicValues The encoded public values.
     * @param _proofBytes   The encoded proof.
     *
     * @return If verification of ZK proof is passed or not.
```

3. Hash of the block C
```

#### Proving Fault

The ZK proof generated by a prover can be verified on chain.

```solidity
    /**
     * @notice Proves that a specific output is invalid using ZK proof.
     *
     * @param _pos          Position of the last valid segment.
     * @param _targetOutput The output root of target block.
     * @param _zkProof      SP1 proofs composed of points and scalars.
     */
    function proveFault(
        uint256 _pos,
        bytes32 _targetOutput,
        uint256[] calldata _zkProof
    ) external;
```

The verify function of `ZKVerifier` contract, which will be used in `proveFault` for verification of provided ZK proof,
will be implemented following the below interface.

```solidity
interface ZKVerifier {
    /** 

>      * @notice The entrypoint for verifying the proof of ZK fault proof.

Reviewer: ```suggestion
     * @notice The entrypoint for verifying the ZK fault proof.
```

     *
     * @param _publicValues The encoded public values.

>      * @param _proofBytes The encoded proof.

Reviewer: ```suggestion
     * @param _proofBytes   The encoded proof.
```
